ament_tools/build_types/ament_python.py

```diff
@@ -369,14 +369,18 @@ def _undo_install(self, context):
         if os.path.exists(install_log):
             with open(install_log, 'r') as h:
                 lines = [l.rstrip() for l in h.readlines()]
-            directories = []
+            directories = set()
+            install_space = context.install_space + os.sep
             for line in lines:
                 if os.path.exists(line) and \
-                        line.startswith(context.install_space):
+                        line.startswith(install_space):
```

Comment:
I don't know if there are cases on Windows where that is the case. Since the code was doing the same before (checking with `startswith`) I don't plan to look into that in the context of this PR.